src/partition/repart.c

```diff
@@ -2256,12 +2262,12 @@ static int device_kernel_partitions_supported(int fd) {
         if (fstat(fd, &st) < 0)
                 return log_error_errno(fd, "Failed to fstat() image file: %m");
         if (!S_ISBLK(st.st_mode))
-                return false;
+                return PARTITIONS_NOT_A_BLOCK_DEVICE;
```

Comment:
it's about logging. i.e. the function logs about all errors. If we make ENOTBLK the thing we return here on error it's a bit ugly, since suddenly there would be one error we don't log about...

but ok, i guess as long as it has a big fat comment that'd be ok.